### DISCUSSION TITLE
Removing background map from rain radar image

### DISCUSSION THREAD

**Discussion by 19217** [ImageMagick/ImageMagick#3013] [Help]
Hello I'm trying to remove the background map from rain radar images so that I can overlay them onto a google map. 
https://i.imgur.com/HrNo8ol.jpg

I can remove most of it and make it transparent using this command 
`convert rain.jpg ( +clone -colorspace HCL -channel G -separate +channel -threshold 6000% ) -alpha off -compose CopyOpacity -composite out.png`
But I have a problem that in thunderstorms the colour white is used and this is also removed because it is similar to the white that's used on the background map
https://i.imgur.com/fIavoSd.png

I don't know much about IM, this command is what someone else used to remove the background map from satellite images. Does anyone know a way to keep the white in thunderstorms? Maybe a way could be to make a one copy of the image that does the command above then another copy that is able to keep the white in thunderstorms by detecting surrounding colours like purple and red and then combine the two images or could there be a better approach?

I have IM version 7.0.10-51 Q16 x64

**Comment by snibgo** [comment]
Perhaps the radar images are available somewhere without the background maps.

For a one-off job, I would do it with Gimp.

Your image has JPEG compression, which makes the task more difficult.

If you have many images with the same background map, you could combine them to make a "blank" map. Then any differences from that map will be radar data.

From just this image, a possible approach is:
```
magick ^
  HrNo8ol.jpg ^
  -colorspace HCL -channel 1 -separate +channel ^
  -threshold 5% -morphology close disk:1.0 ^
  x2.png
```
![x2](https://user-images.githubusercontent.com/33812368/104017394-f7981d00-51af-11eb-8218-d2a07b38e92a.png)

x2.png is then a mask: white for radar, black for background. It is noisy, and could be improved, but a better source would be even better.
